feat: display `alignment` & `criteria` for Open Badges 3.0

Oran-Dan opened the pull request:
# Description of change
I've added the alignment field in Openbadge to the fields which are displayed in the Badge Overview.

The `serde_json` dependency updated from 1.0.20 to 1.0.28. This created a flaky error message being returned in the 
`fn validate_domain_linkage` (meaning it wasn't consistent between different runs and different environments).
To counter this we have made static error messages which are more explicit.

## Links to any relevant issues
- #385 

## How the change has been tested
Ran program and checked visually.
Existing tests pass.

## Definition of Done checklist
Add an `x` to the boxes that are relevant to your changes.

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

daniel-mader committed:
ci: release version 0.6.12 (#370)

daniel-mader commented on Cargo.lock:
Is there a specific reason why this dependency was updated (since no other Rust files changed)?

Oran-Dan replied:
This is an automatic version bump since the toml file only specifies serde_json = 1.0, so the third part automatically went from .20 to .28. Other than that I also discussed this with Nander already, it's fine

daniel-mader commented on .github/workflows/format-lint-test.yaml:
```diff
@@ -8,7 +10,7 @@ on:
 jobs:
   unime_frontend:
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-22.04
```
This doesn't need to be pinned anymore since `ubuntu-latest` seems to be fixed upstream.

Oran-Dan replied:
fixed

daniel-mader commented on identity-wallet/src/state/did/validate_domain_linkage.rs:
```diff
@@ -131,10 +131,15 @@ async fn fetch_configuration(mut url: url::Url) -> Result<DomainLinkageConfigura
-    let response = reqwest::get(url).await.map_err(|e| e.to_string())?;
+    let response = reqwest::get(url)
+        .await
+        .map_err(|_| "failed to get response from resource url".to_string())?;
```
Should we add the actual URL to the error message to help debugging?

Oran-Dan replied:
fixed

daniel-mader commented on unime/src/i18n/en/index.ts:
```diff
@@ -268,7 +268,7 @@ const en = {
       ISSUED_BY: 'Issued by',
       DESCRIPTION: 'Description',
-      CONTENTS: 'Contents',
+      CRITERIA: 'Criteria',
```
**CONTENTS** was a general-purpose translation, whereas **CRITERIA** is specific to Open Badges 3.0. We should therefore wrap it inside a **OPEN_BADGES** object to keep it scoped.

```
DETAILS: {
  OPEN_BADGES: {
    CRITERIA: 'Criteria'
  }
}
```

Oran-Dan replied:
fixed

daniel-mader commented on unime/src/i18n/de-DE/index.ts:
```diff
@@ -269,7 +269,7 @@ const de = {
       ISSUED_BY: 'Ausgestellt von',
       DESCRIPTION: 'Beschreibung',
-      CONTENTS: 'Inhalt',
+      CRITERIA: 'Criteria',
```
Translates to "Kriterien".

Oran-Dan replied:
fixed

daniel-mader commented on unime/src/routes/credentials/[id]/OpenBadgeRenderer.svelte:
```diff
@@ -24,14 +24,25 @@
-  <TextFieldRenderer key={'Achievement name'} value={credential.data.credentialSubject?.achievement?.name} />
+  {#if credential.data.credentialSubject?.achievement?.alignment?.length > 0}
+    <!-- Loop over the alignment array and render each item with a subtle dividing line -->
+    {#each credential.data.credentialSubject.achievement.alignment as alignItem}
```
`alignItem` -> `alignmentItem`

Oran-Dan replied:
fixed

maiertech commented on unime/src/i18n/de-DE/index.ts:
```diff
@@ -269,7 +269,7 @@ const de = {
       VALID: 'Gültig',
       ISSUED_BY: 'Ausgestellt von',
```
No related to your PR: I think this should be 'Ausgestellt durch'.

Oran-Dan replied:
fixed

maiertech reviewed (commented):
For you next frontend PR, please add some hints for reviewers on how to test the UI changes. In this case I can't fire up my local UniCore because it does not have the credential config I need to test. I figured it out, but the easier you make it for me the faster I review :)

daniel-mader commented on unime/src/routes/credentials/[id]/OpenBadgeRenderer.svelte:
```diff
@@ -24,14 +24,25 @@
+    {/each}
+  {/if}
+
+  <TextFieldRenderer key={'Achievement type'} value={credential.data.credentialSubject?.achievement?.achievementType} />
```
`achievementType` is not mandatory. Please render conditionally (only if not undefined).

Oran-Dan replied:
fixed

dependabot[bot] committed:
chore(deps-dev): bump globals from 15.9.0 to 15.11.0 (#389)

Bumps [globals](https://github.com/sindresorhus/globals) from 15.9.0 to 15.11.0.
- [Release notes](https://github.com/sindresorhus/globals/releases)
- [Commits](https://github.com/sindresorhus/globals/compare/v15.9.0...v15.11.0)

---
updated-dependencies:
- dependency-name: globals
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): bump typescript-eslint from 8.2.0 to 8.12.0 (#391)

Bumps [typescript-eslint](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/typescript-eslint) from 8.2.0 to 8.12.0.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/typescript-eslint/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v8.12.0/packages/typescript-eslint)

---
updated-dependencies:
- dependency-name: typescript-eslint
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): bump prettier-plugin-svelte from 3.2.6 to 3.2.7 (#382)

Bumps [prettier-plugin-svelte](https://github.com/sveltejs/prettier-plugin-svelte) from 3.2.6 to 3.2.7.
- [Changelog](https://github.com/sveltejs/prettier-plugin-svelte/blob/master/CHANGELOG.md)
- [Commits](https://github.com/sveltejs/prettier-plugin-svelte/compare/v3.2.6...v3.2.7)

---
updated-dependencies:
- dependency-name: prettier-plugin-svelte
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): bump postcss from 8.4.45 to 8.4.47 (#375)

Bumps [postcss](https://github.com/postcss/postcss) from 8.4.45 to 8.4.47.
- [Release notes](https://github.com/postcss/postcss/releases)
- [Changelog](https://github.com/postcss/postcss/blob/main/CHANGELOG.md)
- [Commits](https://github.com/postcss/postcss/compare/8.4.45...8.4.47)

---
updated-dependencies:
- dependency-name: postcss
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Oran-Dan commented:
> For you next frontend PR, please add some hints for reviewers on how to test the UI changes. In this case I can't fire up my local UniCore because it does not have the credential config I need to test. I figured it out, but the easier you make it for me the faster I review :)

You're right, duely noted

Oran-Dan committed:
chore: resolve PR comments

Oran-Dan committed:
Merge branch 'dev' into feat/display_alignment_obv3

Oran-Dan committed:
chore: resole PR comment

Oran-Dan committed:
chore: resolve PR comment

Oran-Dan committed:
Merge branch 'ci/release-0.7' into feat/display_alignment_obv3
